docs/components/table-data.md

```diff
@@ -0,0 +1,479 @@
+# Table Data
+
+Component that takes care of fetching and rendering data in a table format.
+
+:::tip NOTE
+If you are looking for a simpler table component that does not integrate data fetching, check out [KTableView](/components/table-view).
+:::
+
+<KTableData
+  :fetcher="basicFetcher"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableData
+  :fetcher="fetcher"
+  :headers="headers"
+/>
+```
+
+## Underlying KTableView Component
+
+KTableData uses KTableView component providing data fetching layer on top. KTableData supports all KTableView [props](/components/table-view#props) (with the exception of `data` prop, which is replaced with [`fetcher` prop](#fetcher) in KTableData), [slots](/components/table-view#slots) and [events](/components/table-view#events) (with the exception of [pagination events](/components/table-view#pagination-events) which are not emitted by KTableData). `headers` and `hidePaginationWhenOptional` props as well as `update:table-preferences` event are handled slightly differently in KTableData. See below sections for details.
+
+## Props
+
+### headers
+
+In addition to all header object properties supported in KTableView, KTableData also allows specifying whether a column should be sortable via a function provided via [`sortHandlerFunction` prop](#sorthandlerfunction).
+
+```ts
+interface TableDataHeader {
+  /** must be unique for each column, see Reserved Header Keys section for more information about 'actions' key value */
+  key: string
+  /** visible column header text */
+  label: string
+  /** this property defines whether sort icon should be displayed next to the column header and whether the column header will emit sort event upon clicking on it */
+  sortable?: boolean
+  /** allow toggling column visibility */
+  hidable?: boolean
+  /** when provided, an info icon will be rendered next to the column label, upon hovering on which the tooltip will be revealed */
+  tooltip?: string
+  /** whether column header text should be hidden (only visible to screen readers) */
+  hideLabel?: boolean
+  /** Whether KTableData should use function passed through sortHandlerFunction prop to apply sorting logic to this column */
+  useSortHandlerFunction?: boolean
+}
+```
+
+For an example of `headers` prop usage please refer to [`fetcher` prop documentation](#fetcher) below.
+
+### fetcher
+
+Function that handles data fetching, server-side search and pagination. It takes a single param as an argument which you can conveniently pass as a prop (see [`initialFetcherParams` prop](#initialfetcherparams) for details).
```

Comment:
suggestion: "server-side search" implies this component does something in SSR or communicates with the server. I think this should be reworded to something that suggests that it supports querying new data from an API.